examples/ai-ml/ray/main.tf

```diff
@@ -0,0 +1,365 @@
+      hostname = var.eks_cluster_domain
+    })]
+    set_sensitive = [
+      {
+        name  = "grafana.adminPassword"
+        value = data.aws_secretsmanager_secret_version.admin_password_version.secret_string
+      }
+    ]
+  }
+
+  tags = local.tags
+}
+
+#---------------------------------------------------------------
+# Deploy Ray Cluster Resources
+#---------------------------------------------------------------
+resource "aws_kms_key" "objects" {
+  enable_key_rotation     = true
+  description             = "KMS key is used to encrypt bucket objects"
+  deletion_window_in_days = 7
+}
+
+
+#tfsec:ignore:aws-s3-enable-bucket-logging tfsec:ignore:aws-s3-enable-versioning
+module "s3_bucket" {
+  source  = "terraform-aws-modules/s3-bucket/aws"
+  version = "v3.3.0"
+
+  bucket_prefix           = "ray-demo-models-"
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+
+  server_side_encryption_configuration = {
+    rule = {
+      apply_server_side_encryption_by_default = {
+        kms_master_key_id = aws_kms_key.objects.arn
+        sse_algorithm     = "aws:kms"
+      }
+    }
+  }
+}
+
+data "aws_iam_policy_document" "irsa_policy" {
+  statement {
+    actions   = ["s3:ListBucket"]
+    resources = ["${module.s3_bucket.s3_bucket_arn}"]
+  }
+  statement {
+    actions   = ["s3:*Object"]
+    resources = ["${module.s3_bucket.s3_bucket_arn}/*"]
+  }
+  statement {
+    actions = [
+      "ssm:PutParameter",
+      "ssm:DeleteParameter",
+      "ssm:GetParameterHistory",
+      "ssm:GetParametersByPath",
+      "ssm:GetParameters",
+      "ssm:GetParameter",
+      "ssm:DeleteParameters",
+      "ssm:DescribeParameters"
+    ]
+    resources = ["*"]
+  }
+}
+
+resource "aws_iam_policy" "irsa_policy" {
+  description = "IAM Policy for IRSA"
+  name_prefix = substr("${module.eks_blueprints.eks_cluster_id}-${local.namespace}-access", 0, 127)
+  policy      = data.aws_iam_policy_document.irsa_policy.json
+}
+
+module "cluster_irsa" {
+  source                     = "../../../modules/irsa"
+  kubernetes_namespace       = local.namespace
+  kubernetes_service_account = "${local.namespace}-sa"
+  irsa_iam_policies          = [aws_iam_policy.irsa_policy.arn]
+  eks_cluster_id             = module.eks_blueprints.eks_cluster_id
+  eks_oidc_provider_arn      = module.eks_blueprints.eks_oidc_provider_arn
+
+  depends_on = [module.s3_bucket]
+}
+
+
+data "kubectl_path_documents" "docs" {
+  pattern = "${path.module}/ray-clusters/*.yaml"
+  vars = {
+    namespace       = local.namespace
+    hostname        = var.eks_cluster_domain == null ? "" : var.eks_cluster_domain
+    account_id      = data.aws_caller_identity.current.account_id
+    region          = local.region
+    service_account = "${local.namespace}-sa"
+  }
+}
+
+# Ugly Terraform issue:
+# See https://github.com/gavinbunney/terraform-provider-kubectl/issues/58#issuecomment-718174177
```

Comment:
You could do a multi-resource doc and just use the one resource - heres an example https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/refactor/v5/examples/karpenter/main.tf#L171-L183